### PR TITLE
feat: [SRM-16104]: Add API for metric graph

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55363)

### CODE REVIEW

**Comment by Varsha-lalwani** [comment]
trigger uts

**Comment by Varsha-lalwani** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -125,6 +125,70 @@ paths:
         id: qz7cxcokqykq6
       tags:
         - ServiceLevelObjective
+  '/v1/orgs/{org}/projects/{project}/metric-graph/{slo-identifier}':
+    parameters:
+      - name: org
```
Comment (mankrit-singh) [inline srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml:130]: could you please use the common defined path params here?

File: srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -430,6 +494,84 @@ components:
...
+      type: object
+      description: Representation of one data point.
+      properties:
+        value:
+          type: number
+          x-stoplight:
+            id: csji6101ywk4d
+          description: Value of the data point.
+          format: double
+        timeStamp:
```
Comment (mankrit-singh) [inline srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml:566]: can this also be snake case?

**Review by mankrit-singh — COMMENTED** [review]
nit comments

**Review by naveen-harness — APPROVED** [review]

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImplTest.java [hunk 3]
```diff
@@ -2580,6 +2597,23 @@ public void testGetSecondaryEventDetails_WithDifferentThreadMessageError() {
         .isInstanceOf(InvalidRequestException.class)
         .hasMessage("All the messages should be of the same thread");
   }
+  @Test
+  @Owner(developers = VARSHA_LALWANI)
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImplTest.java:2601]: we should split this class into smaller classes based on functionality, 2600 lines is already big enough

File: srm-service/chart/templates/ingress.yaml [hunk 4]
```diff
@@ -76,6 +76,13 @@ spec:
                   number: {{ $.Values.service.port }}
             path: /(v1/orgs/.+/projects/.+/slo.*)
             pathType: ImplementationSpecific
+          - backend:
+              service:
+                name: cv-nextgen
+                port:
+                  number: {{ $.Values.service.port }}
+            path: /(v1/orgs/.+/projects/.+/metric-graph/.+)
```
Comment (n00bitax) [inline srm-service/chart/templates/ingress.yaml:84]: do we now need to add ingress rules for every new endpoint?

File: srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (Varsha-lalwani) [reply srm-service/modules/cv-nextgen-service/contracts/openapi/v1/openapi.yaml:130]: I have added this in common defined path Param

File: srm-service/chart/templates/ingress.yaml (hunk 4 above)
Comment (Varsha-lalwani) [reply srm-service/chart/templates/ingress.yaml:84]: yes.

**Review by naveen-harness — APPROVED** [review]

**Review by n00bitax — APPROVED** [review]

**Pull request merged**